### PR TITLE
Add keyboard edinburgh41

### PR DESCRIPTION
**Pull request by schwarzer-geiger**
Initial PR for edinburgh41

## Description
Created and tidied the qmk firmware for edinburgh41.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/edinburgh41/thumbstick.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+bool tapping_mode = false;
+
+// tracks if thumbstick was released
+bool returned_to_zero = true;
+
+// tracks how many times mouse_report.x/y have been read zero in succession
+int zero_reads = 0;
+
+// manipulate mouse report based on current mode
+report_mouse_t pointing_device_task_user(report_mouse_t mouse_report) {
```
Comment (drashna) [inline keyboards/edinburgh41/thumbstick.c:33]: These should not be using user functions.  It should be using kb functions, with the user call, or this should be moved entirely to the keymap.c files

File: keyboards/edinburgh41/thumbstick.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        }
+        mouse_report.x = 0;
+        mouse_report.y = 0;
+    }
+
+    return mouse_report;
+}
+
+// set mode depending on layer
+layer_state_t layer_state_set_user(layer_state_t state) {
```
Comment (drashna) [inline keyboards/edinburgh41/thumbstick.c:77]: same

File: keyboards/edinburgh41/thumbstick.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// import thumbstick settings from config
+#pragma once
+
+#include "config.h"
```
Comment (drashna) [inline keyboards/edinburgh41/thumbstick.c:20]:
```suggestion
#include "edinburgh41.h"
```

File: keyboards/edinburgh41/thumbstick.c [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+bool cursor_mode = false;
+bool scrolling_mode = false;
+bool tapping_mode = false;
+
+// tracks if thumbstick was released
+bool returned_to_zero = true;
+
+// tracks how many times mouse_report.x/y have been read zero in succession
+int zero_reads = 0;
```
Comment (drashna) [inline keyboards/edinburgh41/thumbstick.c:30]:
```suggestion
uint16_t zero_reads = 0;
```

File: keyboards/edinburgh41/config.h [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+#define TAPPING_LAYER 2
+
+#define THUMBSTICK_RIGHT_TAP KC_RIGHT
+#define THUMBSTICK_LEFT_TAP KC_LEFT
+#define THUMBSTICK_UP_TAP KC_UP
+#define THUMBSTICK_DOWN_TAP KC_DOWN
+
+#define CURSOR_SPEED 1
+#define SCROLL_SPEED 0.1
+
```
Comment (drashna) [inline keyboards/edinburgh41/config.h:48]: These are also defined at the keymap level, which will cause issues if they're different values. 

These should be moved to post_config.h, and "checked".  Eg:
```c
#ifndef SCROLLING_LAYER
#    define SCROLLING_LAYER 1
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/edinburgh41/keymaps/lalit/keymap.c [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "thumbstick.c"
```
Comment (fauxpark) [inline keyboards/edinburgh41/keymaps/lalit/keymap.c:18]:
```suggestion
```
Do not include .c files. Make a header and add the .c file to SRC, or put it all in keymap.c.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/edinburgh41/keymaps/lalit/keymap.c (hunk 6 above)
Comment (schwarzer-geiger) [reply keyboards/edinburgh41/keymaps/lalit/keymap.c:18]: Dear fauxpark,
what do you mean exactly by making a header? What I currently did is define all the thumbstick parameters in the config.h, include the config.h in the thumbstick.c file and added thumbstick.c to SRC in rules.mk. Unfortunately, the functionality defined in thumbstick.c is not performed. What should I do exactly?

Many thanks,
Lalit

Comment (fauxpark) [reply keyboards/edinburgh41/keymaps/lalit/keymap.c:18]: The problem with including `.c` files is that if you include it in multiple other `.c` files, you will get multiple copies of the same things declared in it. Standard practice is to create a `.h` file that simply declares the function prototypes, while the `.c` file contains the implementations. All the keymap.c needs to know is that those symbols exist, and the compiler and linker are smart enough to pull it all together.

Comment (schwarzer-geiger) [reply keyboards/edinburgh41/keymaps/lalit/keymap.c]: Dear reviewers,
sadly I'm still struggling with the importing and I was wondering whether you could help me out again. I updated my edinburgh41 repository in my dev_branch to reflect my current version. The structure is currently:
Rules.mk has thumbstick.c added to SRC.
thumbstick.c imports config.h and post_config.h which define parameters used in thumbstick.c.
In my understanding thumbstick.c gets compiled and has access to config.h, so everything should work. However, what actually happens at the moment is:
The firmware compiles without errors, but the thumbstick simply doesn't respond to actions that rely on the parameters defined in config.h. I had this all working in the first PR, but the import structure wasn't to QMK standards, so now I'm trying to get it working and to standard. So I'd be really grateful if somebody could have a quick look and see how I can get this working. Please let me know if you need any more info or explanation.
Many thanks!

Comment (drashna) [reply keyboards/edinburgh41/keymaps/lalit/keymap.c]: For importing C filse, you want to add `SRC += filename.c` to the rules.mk.  This adds it to the list of compile files.

File: keyboards/jakob33/readme.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# jakob33
+
+![jakob33](imgur.com image replace me!)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [Lalit Mistry](https://github.com/schwarzer-geiger)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (fauxpark) [inline keyboards/jakob33/readme.md:9]: Please fill this out.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/jakob33/readme.md (hunk 7 above)
Comment (schwarzer-geiger) [reply keyboards/jakob33/readme.md:9]: Thanks, I don't want to bring any of this yet to the official qmk repo.

Comment (schwarzer-geiger) [reply keyboards/jakob33/readme.md:9]: So at this point my PR is only in relation to the edinburgh41 folder. Can you handle that? I know this is not an ideal way of doing it and I should instead commit only the relevant folder. I am currently trying to figure out how to do that, I'm learning.

**Pull request closed without merging**